kustomize/provider.go

```diff
@@ -136,3 +142,48 @@ func getClientConfig(data []byte, context string) (*rest.Config, error) {
 
 	return clientConfig.ClientConfig()
 }
+
+func newCachedGroupVersionKind(cs *kubernetes.Clientset) cachedGroupVersionKind {
+	cache := cache.New(1*time.Minute, 1*time.Minute)
+
+	return cachedGroupVersionKind{
+		cs:    cs,
+		cache: cache,
+	}
+}
+
+type cachedGroupVersionKind struct {
+	cs    *kubernetes.Clientset
+	cache *cache.Cache
+}
+
+const APIGroupResourcesCacheKey string = "restmapper.GetAPIGroupResources"
+
+func (c cachedGroupVersionKind) getGVR(gvk k8sschema.GroupVersionKind, refreshCache bool) (gvr k8sschema.GroupVersionResource, err error) {
+	var agr []*restmapper.APIGroupResources
+
+	cachedAgr, found := c.cache.Get(APIGroupResourcesCacheKey)
+	if found {
+		agr = cachedAgr.([]*restmapper.APIGroupResources)
+	}
+
+	if found == false || refreshCache == true {
+		agr, err = restmapper.GetAPIGroupResources(c.cs.Discovery())
+		if err != nil {
+			return gvr, fmt.Errorf("discovering API group resources failed: %s", err)
+		}
+		c.cache.Set(APIGroupResourcesCacheKey, agr, cache.DefaultExpiration)
+	}
+
+	rm := restmapper.NewDiscoveryRESTMapper(agr)
```

Comment:
Probably not a problem literally, but in case there is anything expensive here, you might need put it into cache too.

Reply:
It's my understanding, this returns the URL path for the specific kind from the list the cached call queries. So I need to run this per resource. My understanding this is done locally, and does not trigger another API request.